projects/plugins/jetpack/modules/masterbar/admin-menu/class-base-admin-menu.php

```diff
@@ -543,6 +544,65 @@ public function is_item_visible( $item ) {
 		return ! isset( $item[4] ) || false === strpos( $item[4], self::HIDE_CSS_CLASS );
 	}
 
+	/**
+	 * Adds a dashboard switcher to the list of screen meta links of the current page.
+	 */
+	public function add_dashboard_switcher() {
+		$menu_mappings = require __DIR__ . '/menu-mappings.php';
+		$screen        = $this->get_current_screen();
+
+		// Let's show the switcher only in screens that we have a Calypso mapping to switch to.
+		if ( empty( $menu_mappings[ $screen ] ) ) {
+			return;
+		}
+		?>
+		<div id="view-link-wrap" class="hide-if-no-js screen-meta-toggle">
+			<button type="button" id="view-link" class="button show-settings" aria-expanded="false"><?php echo esc_html_x( 'View', 'View options to switch between', 'jetpack' ); ?></button>
+		</div>
+		<div id="view-wrap" class="screen-options-tab__wrapper hide-if-no-js hidden" tabindex="-1">
+			<div class="screen-options-tab__dropdown" data-testid="screen-options-dropdown">
+				<div class="screen-switcher">
+					<a class="screen-switcher__button" href="<?php echo esc_url( add_query_arg( 'preferred-view', 'default' ) ); ?>" data-view="default">
+						<strong><?php esc_html_e( 'Default view', 'jetpack' ); ?></strong>
+						<?php esc_html_e( 'Our WordPress.com redesign for a better experience.', 'jetpack' ); ?>
+					</a>
+					<button class="screen-switcher__button"  data-view="classic">
+						<strong><?php esc_html_e( 'Classic view', 'jetpack' ); ?></strong>
+						<?php esc_html_e( 'The classic WP-Admin WordPress interface.', 'jetpack' ); ?>
+					</button>
+				</div>
+			</div>
+		</div>
+		<?php
+	}
+
+	/**
+	 * Adds a script to append the dashboard switcher to screen meta
+	 */
+	public function dashboard_switcher_scripts() {
+		wp_add_inline_script(
+			'common',
+			"(function( $ ) {
```

Comment:
I'd swear I already gave feedback about avoiding jQuery here, but the comment seems to be missing now 🤔 

Anyway, I think it'd be preferable to write this script using vanilla JS since there is an ongoing effort by 55Cj4-p2 to remove jQuery from `wpcom`.